core/src/main/java/org/apache/carbondata/core/datastore/impl/DefaultFileTypeProvider.java

```diff
@@ -110,6 +110,8 @@ public CarbonFile getCarbonFile(String path, Configuration conf) {
         return new AlluxioCarbonFile(path);
       case VIEWFS:
         return new ViewFSCarbonFile(path);
+      case HDFS_LOCAL:
+        return new HDFSCarbonFile(path);
```

Comment:
a. No need pass the conf ?
b. If required to pass the conf,  please use fall through switch syntax for identical values.
      case HDFS:
      case HDFS_LOCAL:
        return new HDFSCarbonFile(path, conf);

Reply:
changed